### PR TITLE
[SA-1925] Docs for MTO custom RBAC

### PR DESCRIPTION
**Pull request by ShahidHussain-srti**

### CODE REVIEW

**Comment by stakater-user** [comment]
@LilShah Image is available for testing. `docker pull nexus-docker-stakater-nexus.apps.preprod.jlvwjls8.kubeapp.cloud/stakater/stakaterclouddocs:SNAPSHOT-PR-96-ede7ad26`

**Comment by stakater-user** [comment]
@LilShah Image is available for testing. `docker pull nexus-docker-stakater-nexus.apps.preprod.jlvwjls8.kubeapp.cloud/stakater/stakaterclouddocs:SNAPSHOT-PR-96-27c610b4`

File: docs/content/sre/multi-tenant-operator/features.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ The major features of Multi Tenant Operator (MTO) are described below.
 
 RBAC is one of the most complicated and error-prone parts of Kubernetes. With Multi Tenant Operator, you can rest assured that RBAC is configured with the "least privilege" mindset and all rules are kept up-to-date with zero manual effort.
 
-Multi Tenant Operator provides several ClusterRoles that are automatically bound to the Tenants Namespaces used for managing access to the Namespaces and the resources they contain. You can also modify  the default roles or create new roles to have full control and customize access control for your users and teams.
+Multi Tenant Operator binds existing ClusterRoles to the Tenants Namespaces used for managing access to the Namespaces and the resources they contain. You can also modify the default roles or create new roles to have full control and customize access control for your users and teams.
```
Comment (MahnoorAsghar) [inline docs/content/sre/multi-tenant-operator/features.md]: nit edit: Tenant's

File: docs/content/sre/multi-tenant-operator/integration-config.md [hunk 2]
```diff
@@ -67,7 +98,74 @@ spec:
 
 Following are the different components that can be used to configure multi-tenancy in a cluster via Multi Tenant Operator.
 
+## TenantRoles
+
+TenantRoles are required within the integration config, as they are used to defined what roles will be applied to each tenant namespace. The field allows optional custom roles, that are then used to create RoleBindings for namespaces that match a labelSelector.
```
Comment (MahnoorAsghar) [inline docs/content/sre/multi-tenant-operator/integration-config.md]: nits: 
- IntegrationConfig (for consistent casing across docs)
- 'as they are used to define', not defined
- 'Tenant' (for consistent casing across docs)

File: docs/content/sre/multi-tenant-operator/integration-config.md [hunk 3]
```diff
@@ -67,7 +98,74 @@ spec:
...
+
+::: warning Note:
+
+If you do not configure roles in any way, then the default OpenShift roles of `owner`, `edit`, and `view` will apply to tenant members. Their details can be found [here](./tenant-roles.html5r)
```
Comment (MahnoorAsghar) [inline docs/content/sre/multi-tenant-operator/integration-config.md]: nit: Tenant (capitalise)

File: docs/content/sre/multi-tenant-operator/integration-config.md [hunk 4]
```diff
@@ -67,7 +98,74 @@ spec:
...
+        - custom-view
+```
+
+### Default
+
+This field contains roles that will be used to create default roleBindings for each namespace that belongs to tenants. These roleBindings are only created for a namespace if that namespaces isn't already matched by the `custom` field below it. Therefore, it is required to have at least one role mentioned within each of its three subfields: `owner`, `editor`, and `viewer`. These 3 subfields also correspond to the member fields of the [Tenant CR](./customresources.html#_2-tenant)
+
+### Custom
+
+An array of custom roles. Similar to the `default` field, you can mention roles within this field as well. However, the custom roles also require the use of a `labelSelector` for each iteration within the array. The roles mentioned here will only apply to these namespaces that are matched by the labelSelector. If a namespace is matched by 2 different labelSelectors, then both roles will apply to it. Additionally, roles can be skipped within the labelSelector. These missing roles are then inherited from the `default` roles field . For example, if the following custom roles arrangement is used:
```
Comment (MahnoorAsghar) [inline docs/content/sre/multi-tenant-operator/integration-config.md]: nit: The roles mentioned here will only apply to *those* or *the* namespaces that are matched by the labelSelector.

File: docs/content/sre/multi-tenant-operator/integration-config.md [hunk 5]
```diff
@@ -67,7 +98,74 @@ spec:
...
+      values:
+        - build
+    matchLabels:
+      stakater.com/kind: dev
+  owner:
+    clusterRoles:
+      - custom-owner
+```
+
+Then the `editor` and `viewer` roles will be taken from the `default` roles field, as that is required to have at least one role mentioned.
```
Comment (MahnoorAsghar) [inline docs/content/sre/multi-tenant-operator/integration-config.md:166]: it is required to have at least one role for ..? (each tenant namespace?)
Finish that thought so newbies like me fully grasp *what* exactly needs at least one editor, viewer and owner role to be defined for it.

File: docs/content/sre/multi-tenant-operator/usecases/custom-roles.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        - key: stakater.com/tenant
+          operator: In
+          values:
+            - bluesky
+      owner:
+        clusterRoles:
+          - admin
+```
+
+New Bindings will be created for the tenant owners of `bluesky` and `alpha`, corresponding to the `admin` Role. Bindings for editors and viewer will be inherited from the `default roles`. All other tenant owners will have an `edit` Role bound to them within their namespaces
```
Comment (MahnoorAsghar) [inline docs/content/sre/multi-tenant-operator/usecases/custom-roles.md]: do capitalise Tenant everywhere :)

File: docs/content/sre/multi-tenant-operator/usecases/integrationconfig.md [hunk 7]
```diff
@@ -2,9 +2,9 @@
 
 Bill is a cluster admin who can use `IntegrationConfig` to configure how `Multi Tenant Operator (MTO)` manages the cluster.
 
-By default, MTO watches all namespaces and will enforce all the governing policies on them. 
-All namespaces managed by MTO require the `stakater.com/tenant` label. 
-MTO ignores privileged namespaces that are mentioned in the IntegrationConfig, and does not manage them. These namespaces do not require the above-mentioned label. 
+By default, MTO watches all namespaces and will enforce all the governing policies on them.
+All namespaces managed by MTO require the `stakater.com/tenant` label.
+MTO ignores privileged namespaces that are mentioned in the IntegrationConfig, and does not manage them. These namespaces do not require the above-mentioned label.
```
Comment (MahnoorAsghar) [inline docs/content/sre/multi-tenant-operator/usecases/integrationconfig.md]: Shall we change the last sentence to: "Any tenant label on such namespaces will be ignored". Does it sound clearer conceptually?

Comment (ShahidHussain-srti) [reply docs/content/sre/multi-tenant-operator/usecases/integrationconfig.md]: can be added

File: docs/content/sre/multi-tenant-operator/integration-config.md (hunk 5 above)
Comment (ShahidHussain-srti) [reply docs/content/sre/multi-tenant-operator/integration-config.md:166]: the default roles field, as mentioned

**Comment by stakater-user** [comment]
@LilShah Image is available for testing. `docker pull nexus-docker-stakater-nexus.apps.preprod.jlvwjls8.kubeapp.cloud/stakater/stakaterclouddocs:SNAPSHOT-PR-96-24f8519a`

**Comment by stakater-user** [comment]
@LilShah Image is available for testing. `docker pull nexus-docker-stakater-nexus.apps.preprod.jlvwjls8.kubeapp.cloud/stakater/stakaterclouddocs:SNAPSHOT-PR-96-68b0f068`

**Comment by stakater-user** [comment]
@LilShah Image is available for testing. `docker pull nexus-docker-stakater-nexus.apps.preprod.jlvwjls8.kubeapp.cloud/stakater/stakaterclouddocs:SNAPSHOT-PR-96-c53bfac6`

**Comment by stakater-user** [comment]
@LilShah Image is available for testing. `docker pull nexus-docker-stakater-nexus.apps.preprod.jlvwjls8.kubeapp.cloud/stakater/stakaterclouddocs:SNAPSHOT-PR-96-33d7e5d1`

**Review by MahnoorAsghar — APPROVED** [review]

**Comment by stakater-user** [comment]
@LilShah Image is available for testing. `docker pull nexus-docker-stakater-nexus.apps.saap-prod.clusters.stakater.io/stakater/stakaterclouddocs:v0.0.74`

**Pull request merged**